source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -295,7 +295,7 @@ absl::Status ProcessorState::handleBodyResponse(const BodyResponse& response) {
           const auto mut_status = MutationUtils::applyHeaderMutations(
               common_response.header_mutation(), *headers_,
               common_response.status() == CommonResponse::CONTINUE_AND_REPLACE,
-              filter_.config().mutationChecker(), filter_.stats().rejected_header_mutations_);
+              filter_.config().mutationChecker(), filter_.stats().rejected_header_mutations_, body_mode_ == ProcessingMode::STREAMED);
```

Comment:
The callback_state is BufferedPartialBodyCallBack, so the body_mode_ here is BUFFERED_PARTIAL.  It probably is easier just set it to false?

Reply:
I think as general principle, we should avoid hardcode value if possible. I agree with using hardcode for immediateResponse and trailer case though

Here, existing code is clean/consistent and less error prone, compared with looking for `callback_state == BufferedPartialBodyCallBack` then hardcoding the value.